test/rules/prefer_final_locals_test.dart

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(PreferFinalLocalsPatternsTest);
+  });
+}
+
+@reflectiveTest
+class PreferFinalLocalsPatternsTest extends LintRuleTest {
+  @override
+  List<String> get experiments => ['patterns', 'records'];
+
+  @override
+  String get lintRule => 'prefer_final_locals';
+
+  test_destructured_listPattern_mutated_ok() async {
+    await assertNoDiagnostics(r'''
+f() {
+  var [a, b, ...rest] = [1, 2, 3, 4, 5, 6, 7];
+  print('${++a}$b$rest');
+}
+''');
+  }
+
+  test_destructured_listPattern_ok() async {
```

Comment:
I'd drop the `_ok` (I think the `assertNoDiagnostics` says enough), especially here where we're expecting a lint.

Reply:
Done in https://github.com/dart-lang/linter/pull/4225